validator/tests/test_scheduler/yaml_generator.py

```diff
@@ -0,0 +1,333 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import argparse
+import os
+import random
+import sys
+
+import yaml
+
+
+TEST_DIR = os.path.abspath(os.path.dirname(__file__))
+DATA_DIR = os.path.join(TEST_DIR, 'data')
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        'file',
+        help="The filepath and name relative to "
+             "validator/tests/test_scheduler/data")
+
+    parser.add_argument(
+        '-s',
+        '--seed',
+        help="The seed to make the random generator reproduceable")
+
+    parser.add_argument(
+        '-a',
+        '--addresses',
+        default=100,
+        help="The number of addresses in total")
+
+    parser.add_argument(
+        '--namespace',
+        default='aaaaaa,.5',
+        help="A namespace, float comma separated pair specifying "
+             "a namespace to use for some portion of the addresses"
+    )
+
+    parser.add_argument(
+        '-w',
+        '--wildcard',
+        default='70,70,70,70',
+        help="A comma separated quad specifying min and max length for inputs "
+             "then outputs")
+
+    parser.add_argument(
+        '-t',
+        '--transactions',
+        default="1,1",
+        help="A comma separated pair of numbers for min and max transactions "
+             "per batch")
+
+    parser.add_argument(
+        '-b',
+        '--batches',
+        default=100,
+        help="The number of batches to produce")
+
+    parser.add_argument(
+        '-v',
+        '--validity',
+        default=1.0,
+        type=float,
+        help="A number between 0 and 1 specifying the probability of invalid "
+             "transactions")
+
+    parser.add_argument(
+        '-i',
+        '--inputs',
+        default='2,3',
+        help="A comma separated pair specifying the minimum and "
+             "maximum number of addresses for inputs")
+
+    parser.add_argument(
+        '-o',
+        '--outputs',
+        default='2,3',
+        help="A comma separated pair specifying the minimum "
+             "and maximum number of addresses for outputs")
+
+    return parser.parse_args(args)
+
+
+class GeneratorCliException(Exception):
+    pass
+
+
+def main_wrapper(args):
+    try:
+        main(args)
+    except GeneratorCliException as e:
+        print(e)
+
+
+def main(args):
+    opts = parse_args(args=args)
+
+    if opts.seed is None:
+        seed = random.randrange(sys.maxsize)
+    else:
+        seed = int(opts.seed)
+    random.seed(seed)
+
+    sentence = generate_cmd_header(opts, seed=seed)
```

Comment:
I was actually referring to `sentence`, but I like the new method name too.

`sentence` -> `cmd_header`

Reply:
Ah cool. Thanks.